tests/unit/credentials_test.py

```diff
@@ -45,11 +45,16 @@ def test_credentials_get(self):
         self.assertEqual(creds.session_token, None)
 
 
-class CredListResponse(object):
+class TokenResponse(object):
```

Comment:
Remove changes in this unit test which doesn't add any value

Reply:
I think all changes here add value. Status and the data here are being used in the first request's response. And we can see it be used.

I suppose we could use the same response for `TokenResponse` and `CredList` but I do think they should be different because they are different interactions.

Reply:
As this test (not your change) doesn't prevent any logical error in the code, there is no point updating it. The right way to test is to run with AWS.

Reply:
I agree that it doesn't have coverage for all of the cases, but the addition does atleast illustrate the pathway for the IMDS route. I can add a comment to the test to make it explicit that it is only testing the happy path, if that'd help?

Reply:
The purpose of unit test is to cover all possible values. That's the reason I do not like such partial tests.

Reply:
Ok, I removed the test for `IamAwsProvider` in https://github.com/minio/minio-py/pull/1437/commits/3a5cd228e85c9c44b9313ef72ef2bc865687aff9.

Reply:
FWIW, I did test it with AWS (atleast the original branch that I had) and that worked well. I can verify the exact code here tomorrow, too.